server/src/auth/auth.service.ts

```diff
@@ -1,15 +1,24 @@
 import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { UsersService } from '../users/users.service';
+import { User } from '../users/entities/user.entity';
 import { JwtService } from '@nestjs/jwt';
 import * as bcrypt from 'bcrypt';
 
+
 @Injectable()
 export class AuthService {
     constructor(private usersService: UsersService, private jwtService: JwtService) {}
 
     async validateUser(email: string, pass: string): Promise<any> {
         // Check if user with email exists in database
-        const user = await this.usersService.findByEmail(email);
+        let user: User;
+        try {
+            user = await this.usersService.findByEmail(email);
+        } catch (err) {
+            err.status = HttpStatus.UNAUTHORIZED
+            err.response.status = HttpStatus.UNAUTHORIZED
+            throw err           
```

Comment:
Definitely can do that. The only reason I have it set up like this:

- Returning supplies this error message `{statusCode: 401, message: "Unauthorized"}` 
- Throwing it allows me to supply a more specific error `{status: 401, error: "Email not found"}`

I could change the client to handle the default error message if you'd prefer to keep it that way.

Reply:
yeah i guess let's not get bogged down in these error messages
but i'd say controlling the error message is a good idea

only caveat i'd say is that `email not found` message might be giving away too much info
ultra hackerz could find out which emails have accounts on our site

Reply:
Okay, I'll keep as is for now for v1 given that it works and other parts of this PR rely on the specific messages. We can come back to this. It would require a decent change to how we display error messages currently (on a material-ui input's error prop/ FormHelperText).